daml-lf/kv-support/src/main/scala/com/daml/lf/kv/contracts/ContractConversions.scala

```diff
@@ -13,16 +13,17 @@ import scala.util.{Failure, Success, Try}
 object ContractConversions {
 
   def encodeContractInstance(
-      coinst: Value.VersionedContractInstance
+      coinst: Value.VersionedContractInstance,
+      agreement: String,
   ): Either[ValueCoder.EncodeError, RawContractInstance] =
     for {
-      message <- TransactionCoder.encodeContractInstance(ValueCoder.CidEncoder, coinst)
+      message <- TransactionCoder.encodeContractInstance(ValueCoder.CidEncoder, coinst, agreement)
       bytes <- SafeProto.toByteString(message).left.map(ValueCoder.EncodeError(_))
     } yield RawContractInstance(bytes)
 
   def decodeContractInstance(
       rawContractInstance: RawContractInstance
-  ): Either[ConversionError, Value.VersionedContractInstance] =
+  ): Either[ConversionError, (Value.VersionedContractInstance, String)] =
```

Comment:
I couldn't find any uses of this function, `encodeContractInstance` or `ContractConversions` anywhere other than `ContractConversionsSpec`, maybe we can just drop these?
If they _are_ used somewhere else, I'd prefer if this returned the agreement inside a "newtype", the way we do for e.g. `TemplateId`.

Reply:
this is used by kv.

Reply:
aah fair enough, I missed that since it's on a different repo. Thanks!